### PR TITLE
Add Spotless

### PR DESCRIPTION
**Pull request by srimanachanta**
uses the config from alllwpilib

### CODE REVIEW

**Comment by ThadHouse** [comment]
You need to copy the git attributes as well from allwpilib, or else spotless breaks on windows machines.

File: .github/workflows/lint-format.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    container: wpilib/ubuntu-base:22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Fetch all history and metadata
+        run: |
+          git config --global --add safe.directory $GITHUB_WORKSPACE
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Run Java format
+        run: ./gradlew spotlessCheck
```
Comment (ThadHouse) [inline .github/workflows/lint-format.yml:22]: New line at end of file needed.

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,5 @@
+*.gradle text eol=lf
+*.java text eol=lf
+*.json text eol=lf
+*.md text eol=lf
+*.xml text eol=lf
```
Comment (ThadHouse) [inline .gitattributes:5]: New line at end of file

Comment (srimanachanta) [reply .gitattributes:5]: No new line in allwpilib

Comment (ThadHouse) [reply .gitattributes:5]: Well there should be. Thats an issue there. All files need one.

Comment (srimanachanta) [reply .gitattributes:5]: Nvm, I read it wrong sry

**Pull request closed without merging**